core/src/main/java/org/apache/iceberg/rest/responses/RESTCatalogConfigResponse.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest.responses;
+
+import java.util.Collections;
+import java.util.Map;
+import org.apache.iceberg.relocated.com.google.common.base.MoreObjects;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.relocated.com.google.common.collect.Maps;
+
+/**
+ * Represents a response to requesting server-side provided configuration for the REST catalog.
+ * This allows client provided values to be overridden by the server or defaulted if not provided by the client.
+ * <p>
+ * This configuration should be fetched with a one time initial HTTP request during the REST catalog initialization.
+ * The connection used to fetch the configuration should then be disposed and a new http client should be
+ * created for use with the catalog.
+ * <p>
+ * Configuration from the server consists of two sets of key/value pairs.
+ * <ul>
+ *   <li> defaults - properties that should be used as default configuration </li>
+ *   <li> overrides - properties that should be used to override client configuration </li>
+ * </ul>
+ */
+public class RESTCatalogConfigResponse {
+
+  private Map<String, String> defaults;
+  private Map<String, String> overrides;
+
+  public RESTCatalogConfigResponse() {
+    // Required for Jackson deserialization
+  }
+
+  private RESTCatalogConfigResponse(Map<String, String> defaults, Map<String, String> overrides) {
+    this.defaults = defaults;
+    this.overrides = overrides;
+    validate();
+  }
+
+  RESTCatalogConfigResponse validate() {
+    return this;
+  }
+
+  /**
+   * Properties that should be used as default configuration. {@code defaults} have the lowest priority
+   * and should be applied before the client provided configuration.
+   *
+   * @return properties that should be used as default configuration
+   */
+  public Map<String, String> defaults() {
+    return defaults != null ? Maps.newHashMap(defaults) : Collections.emptyMap();
+  }
```

Comment:
Make a defensive copy, as the `defaults` are merged with `overrides` and we don't want anybody mutating them incorrectly in the process.